src/go/configgenerator/listener_generator.go

```diff
@@ -835,3 +857,11 @@ func makeRouterFilter(opts options.ConfigGeneratorOptions) *hcmpb.HttpFilter {
 	}
 	return routerFilter
 }
+
+func parseDepErrorBehavior(stringVal string) (commonpb.DependencyErrorBehavior, error) {
+	depErrorBehaviorInt, ok := commonpb.DependencyErrorBehavior_value[stringVal]
```

Comment:
I feel very strange to use proto generated enum directly.  prefer to use string as flag and variable,  and convert the string to proto enum when filling the proto.

Reply:
That is what is happening, right? The flag and option are declared as string. This will parse the string option into the proto enum, which is then filled in the filter config.

Reply:
I see.  You are using the proto map to to the convert.  OK, then.